src/Navigations/Router.js

```diff
@@ -0,0 +1,19 @@
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+
+import Users from "../components/users/Users";
+import UsersDetails from "../components/details/userDetails";
+import UserPosts from "../components/posts/userPosts";
+
+const Router = () => {
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Users />}></Route>
+        <Route path="details/:userId" element={<UsersDetails />}></Route>
+        <Route path="details/:userId/posts" element={<UserPosts />}></Route>
```

Comment:
Add a not-found page and direct the user there when any of the these routes do not matches